bitcoin/src/psbt/serialize.rs

```diff
@@ -225,7 +225,7 @@ impl Serialize for KeySource {
 impl Deserialize for KeySource {
     fn deserialize(bytes: &[u8]) -> Result<Self, Error> {
         if bytes.len() < 4 {
-            return Err(encode::Error::from(io::Error::from(io::ErrorKind::UnexpectedEof)).into())
+            return Err(io::Error::from(io::ErrorKind::UnexpectedEof).into())
```

Comment:
This style is a bit confusing ("WTF?, this isn't supposed to return `io::Error`") but not a big deal.